src/mmw/js/src/core/models.js

```diff
@@ -61,36 +61,89 @@ var TaskModel = Backbone.Model.extend({
         }
     },
 
+    // Cancels any currently running jobs. The promise returned
+    // by previous calls to pollForResults will be rejected.
+    reset: function() {
+        this.set({
+            'job': null,
+            'result': null,
+            'status': null
+        });
+    },
+
+    // taskHelper should be an object containing an optional object,
+    // postData, an optional function, onStart, and functions pollSuccess,
+    // pollFailure, and startFailure.
+    start: function(taskHelper) {
```

Comment:
For "options" args one good convention is to provide defaults. Consider adding:

    taskHelper = _.defaults(taskHelper, {
        onStart: _.identity,
        pollForSuccess: _.identity,
        ...
    });